lib/pages/chats/chat_model.dart

```diff
@@ -0,0 +1,21 @@
+class GroupmeChat {
+  String threadId;
+  String userId;
+  String name;
+  String avatarUrl;
+  String lastMessage;
+  int timestamp;
+
+  GroupmeChat({this.threadId, this.userId, this.name, this.timestamp, this.lastMessage, this.avatarUrl});
+
+  factory GroupmeChat.fromJson(Map<String, dynamic> json) {
+    return GroupmeChat(
+      threadId: json['last_message']['conversation_id'],
+      userId: json['other_user']['id'],
+      name: json['other_user']['name'],
+      avatarUrl: json['other_user']['avatar_url'],
+      lastMessage: json['last_message']['text'],
+      timestamp: json['updated_at']
+    );
+  }
```

Comment:
Based on the [flutter docs](https://flutter.dev/docs/development/data-and-backend/json) I don't think that's what it wants...?

```suggestion
  GroupmeChat.fromJson(Map<String, dynamic> json)
      : threadId = json['last_message']['conversation_id'],
        userId = json['other_user']['id'],
        name = json['other_user']['name'],
        avatarUrl = json['other_user']['avatar_url'],
        lastMessage = json['last_message']['text'],
        timestamp = json['updated_at'];
```

Reply:
This was based on the Photo example from flutter cookbook: https://flutter.dev/docs/cookbook/networking/background-parsing#create-a-photo-class

And I got an error if I didn't put factory in there

Reply:
... ¯\(°_o)/¯

seems like two thirds of the fromJson examples on the internet are constructors and the other third are factories

if it works it works